[ST] Change fields of the YAML files to make more sense and add README for upgrade/downgrade tests (#11698)
strimzi/strimzi-kafka-operator: systemtest/src/test/resources/upgrade/README.md

# Upgrade & downgrade tests

> This short README describes how the upgrade and downgrade tests are configured and what needs to be updated in order to make the tests work.

Reviewer: ```suggestion
This short README describes how the upgrade and downgrade tests are configured and what needs to be updated to make the tests work.
```

  - in case that new Strimzi version supports new Kafka version, you should update the `deployKafkaVersion` as well
- `BundleUpgrade.yaml`
  - `fromVersion: 0.47.0`
  - `fromExamples: strimzi-0.47.0`
  - `fromUrl: https://github.com/strimzi/strimzi-kafka-operator/releases/download/0.47.0/strimzi-0.47.0.zip`
  - `fromKafkaVersionsUrl: https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/0.47.0/kafka-versions.yaml`
- `OlmUpgrade.yaml`
  - `fromVersion: 0.47.0`
  - `fromOlmChannelName: strimzi-0.47.x`
  - `fromExamples: strimzi-0.47.0`
  - `fromUrl: https://github.com/strimzi/strimzi-kafka-operator/releases/download/0.47.0/strimzi-0.47.0.zip`
  - `fromKafkaVersionsUrl: https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/0.47.0/kafka-versions.yaml`

Technically, you can just replace the previous version of Strimzi to the new one - don't forget to change the OLM channel name in `fromOlmChannelName` when doing the search and replace with older version.

## New Kafka version

If you are just adding new Kafka version that is the latest one, you will need to change version of Kafka in `BundleUpgrade.yaml` - inside the `imagesAfterOperations.kafka`.

> If it's just patch release of the older minor version of Kafka supported by Strimzi, you don't need to do anything.

Reviewer: ```suggestion
If it's just a patch release of an already supported minor Kafka version, no changes are needed.
```

> Only once a new version of Strimzi supporting this new version of Kafka, you should change it in the `deployKafkaVersion` field in `BundleDowngrade.yaml`.

Reviewer: ```suggestion
Once a new version of Strimzi supports the new Kafka version, update the deployKafkaVersion field in BundleDowngrade.yaml.
```

## Removal of Kafka version

> If there is Kafka version being removed from the support matrix of Strimzi, you has to change the `deployKafkaVersion` field in `BundleUpgrade.yaml`.

Reviewer 2: ```suggestion
If there is Kafka version being removed from the support matrix of Strimzi, you have to change the `deployKafkaVersion` field in `BundleUpgrade.yaml`.
```
